int32 pwm1
int32 pwm2
int32 pwm3
int32 pwm4
int32 pwm5
int32 pwm6
int32 pwm7